app/uk/gov/hmrc/excisemovementcontrolsystemapi/connectors/MovementMessageConnector.scala

```diff
@@ -16,44 +16,63 @@
-import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpResponse}
+import uk.gov.hmrc.excisemovementcontrolsystemapi.models.auth.DataRequest
+import uk.gov.hmrc.excisemovementcontrolsystemapi.models.eis.Header.EmcsSource
+import uk.gov.hmrc.excisemovementcontrolsystemapi.models.eis._
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient}
 
+import java.nio.charset.StandardCharsets
+import java.time.LocalDateTime
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
 class MovementMessageConnector @Inject()
 (
   httpClient: HttpClient,
   eisUtils: EisUtils,
-  appConfig: AppConfig
+  appConfig: AppConfig,
+  metrics: Metrics
 )(implicit ec: ExecutionContext) extends Logging {
 
-  def post(message: String, messageType: String)(implicit hc: HeaderCarrier): Future[Either[EISErrorResponse, EISResponse]] = {
 
-    //todo: add metrics
+  def submitExciseMovement(request: DataRequest[_], messageType: String)(implicit hc: HeaderCarrier): Future[Either[Result, EISResponse]] = {
+
+    val timer = metrics.defaultRegistry.timer("emcs.eiscontroller.timer").time()
+
     //todo: add retry
-    //todo: message need to be encode Base64
-    val eisRequest = EISRequest(eisUtils.generateCorrelationId, eisUtils.getCurrentDateTimeString, messageType, "APIP", "user1", message)
-
-    httpClient.POST[EISRequest, HttpResponse](
-      appConfig.emcsReceiverMessageUrl,
-      eisRequest,
-      Header.build(eisRequest.emcsCorrelationId, eisRequest.createdDateTime)
-    ).map(
-      response => {
-        response.status match {
-          case OK => Right(Json.parse(response.body).as[EISResponse])
-          case _ =>
-            val errorResponse = Json.parse(response.body).as[EISErrorResponse]
-            logger.error(s"EIS errorResponse. Status: ${errorResponse.status}, message: ${errorResponse.message} and correlationId: ${errorResponse.emcsCorrelationId}")
-            Left(errorResponse)
-        }}
-      )
+    val correlationId = eisUtils.generateCorrelationId
+    val createdDateTime = eisUtils.getCurrentDateTimeString
+    val encodedMessage = eisUtils.createEncoder.encodeToString(request.body.toString.getBytes(StandardCharsets.UTF_8))
+    val eisRequest = EISRequest(correlationId, createdDateTime, messageType, EmcsSource, "user1", encodedMessage)
+    val consignorId = request.consignorId
+
+      httpClient.POST[EISRequest, Either[Result, EISResponse]](
+        appConfig.emcsReceiverMessageUrl,
+        eisRequest,
+        Header.build(correlationId, createdDateTime)
+      )(EISRequest.format, EISHttpReader(correlationId, consignorId, createdDateTime), hc, ec)
+        .andThen { case _ => timer.stop() }
+        .recover {
+          case ex: Throwable =>
+
+            logger.warn(EISErrorMessage(createdDateTime, consignorId, ex.getMessage, correlationId, messageType), ex)
+
+            val error = EISErrorResponse(
```

Comment:
This exception happens when we cannot parse the json to a EISResponse object. Are we ok to return the below json structure to the 3rd party?